### DISCUSSION TITLE
RaspberryMatic OCI Kaltstart - subnet geändert

### DISCUSSION THREAD

**Discussion by TTBahn** [OpenCCU/OpenCCU#2385] [🇩🇪 Fragen&Antworten]
Ich bin von RapberryMatic auf RaspberryMatic Docker/OCI umgestiegen. 
Mit der Installationsanleitung klappte die Installation ohne Probleme, danke für die tolle Arbeit.  Mit einem Backup aus der "alten" Installation war ich sofort wieder up to date. 
Der eigentliche Hintergrund der Neuinstallation war, dass ich iob und die RaspberryMatic auf einem Raspi laufen lassen will.

Wie gesagt nach der frischen RaspberryMatic OCI Installation geht alles Raspberrymatic und iob sehen sich und  können gesteuert werden.
Nach einem Zentralenkaltneustart mittels Herunterfahren oder Spannungsverlust und erneuter Spannungszufuhr findet iob die RaspberryMatic nicht mehr. Die Instanzen hm-rega und hm-rpc wechseln zwischen rot und gelb

Abhilfe: Den RaspberryMatic Container per 
wget -qO - https://raspberrymatic.de/install-docker.sh | sudo bash - 
neu aufsetzen und alles geht wieder.

<IP-ADDRESS>     192.168.1.200  ändert sich nach Kaltstart nicht
<AUX-ADDRESS> 192.168.1.201  ändert sich nach Kaltstart nicht
<INTERFACE>       eth0                 ändert sich nach Kaltstart nicht
<SUBNET>           192.168.1.0/24 nach Kaltstart wird 192.168.1.100/24 vorgeschlagen
<GATEWAY>        192.168.1.1       ändert sich nach Kaltstart nicht
Host rpi                192.168.1.100

RaspberryMatic Version
3.69.7.20230626

Basisplattform 
rpi4 4GB 
OS:Raspberry PiOS lite (64bit)

HomeMatic/homematicIP Funkmodul
RPI-RF-MOD

**Comment by jens-maus** [comment]
Bitte genau beschreiben was du als `<IP-ADDRESS>`, `<AUX-ADDRESS>`, usw. bwi nutzung des install-docker.sh skriptes gesetzt hast.

**Comment by TTBahn** [comment]
Hallo,

mit einer so schnellen Antwort hatte ich nicht gerechnet. Danke.

Ich habe das install-docker.sh und zwischendurch auch die manuelle Installation docker pull genutzt. Im Ergebnis gab es immer den beschrieben Effekt.

Folgen Eingaben habe ich bei der Installation gemacht: 
192.168.1.200 = IP-ADDRESS      wird von install-docker.sh mit 192.168.1. vorgeschlagen
192.168.1.201 = AUX-ADDRESS  wird von install-docker.sh mit 192.168.1. vorgeschlagen
eth0 = INTERFACE                       wird von install-docker.sh richtig erkannt
192.168.1.0/24 = SUBNET           wird von install-docker.sh mit 192.168.1.100/24  vorgeschlagen
192.168.1.1 = GATEWAY              wird von install-docker.sh richtig erkannt; 
192.168.1.100 =Host rpi4            wurde von der Fritzbox per DHCP vergeben; 

Der DHCP Bereich in der Fritbox geht von 192.168.1.50 bis 192.168.1.150

**Comment by TTBahn** [comment]
Hallo zusammen,

da ich da Problem mit der Verbindung iob und ccu nach dem Kaltstart des Host nicht gelöst bekommen habe, habe ich jetzt eine praktische Lösung gefunden. Sicher nicht elegant aber es funktioniert.

Lösung: Ich habe ein kleines Script geschrieben welches die install-docker.sh (von Jens Maus) aufruft und danach den Container startet.
Es müssen nur die mit "" definierten Variablen (z.B CCU_CONTAINER_IP) mit den entsprechenden Adressen in der install-docker.sh gefüllt werden. (Zeile 30-46) Dadurch ist beim automatischen Ablauf des install-docker.sh Script keine Eingabe notwendig.

Z.B Zeile 30
IP address of the container interface
: "${CCU_CONTAINER_IP:=""}" ändern in z.B.  : "${CCU_CONTAINER_IP:="192.168.1.250"}"


Script: host-neustart-container-neu.sh
/home/install-docker.sh
docker start ccu

Dieses kleine Script wird dann automatisch per crontab @reboot aufgerufen.
